src/infos.rs

```diff
@@ -1,5 +1,25 @@
-/// Expose the API_LEVEL
-#[link_section = ".ledger.api_level"]
+const MAX_METADATA_STRING_SIZE: usize = 32;
+
+const fn copy_str_to_u8_array(input: &str) -> [u8; MAX_METADATA_STRING_SIZE] {
+    let mut result = [0u8; MAX_METADATA_STRING_SIZE];
+    let bytes = input.as_bytes();
+    let mut count = 0u32;
+
+    // Ensure we don't copy more than MAX_METADATA_STRING_SIZE
+    let min_length = core::cmp::min(bytes.len(), MAX_METADATA_STRING_SIZE - 1);
+    // Copy bytes to array
+    loop {
+        if count >= min_length as u32 {
+            result[count as usize] = '\n' as u8;
+            break;
+        }
+        result[count as usize] = bytes[count as usize];
+        count += 1;
+    }
+    result
+}
+
+/// Expose the API_LEVEL to the app
 #[used]
 static API_LEVEL: u8 = if cfg!(target_os = "nanos") {
```

Comment:
Should be update to with proper value knwon at build time ?

Reply:
Done in https://github.com/LedgerHQ/ledger-device-rust-sdk/pull/96